### DISCUSSION TITLE
How do I share an object between pages (Route)?

### DISCUSSION THREAD

**Discussion by Miuler** [leptos-rs/leptos#1521] [Q&A]
I have a Router that manages a page managed by a `<Search>` component with er path `"/search"` in which it calls backend to return a list of users that is painted in a table, so far so good, that table is created with a `<a href="/user/id123">` but I'm not sure how is the best way to send the user object to the user detail page managed by the `<UserDetail>` component ?

At the moment **I am removing the `href` from the `<a>` and add an `on:click` event** in which I put the user in the global state, use the `use_navigate` to redirect, and then in the `<UserDetail>` component I retrieve it. Is this the best way? Is there another way?

**Comment by mahadeer** [comment]
Did you try creating a context for holding the currently clicked user and try to drop the data on coming back to the /search page again.
